programs/drift/src/state/perp_market.rs

```diff
@@ -728,18 +732,136 @@ impl Default for AMM {
             oracle_source: OracleSource::default(),
             last_oracle_valid: false,
             target_base_asset_amount_per_lp: 0,
+            per_lp_base: 0,
             padding1: 0,
+            padding2: 0,
             total_fee_earned_per_lp: 0,
             padding: [0; 32],
         }
     }
 }
 
 impl AMM {
+    pub fn get_per_lp_base_unit(self) -> DriftResult<i128> {
+        let scalar: i128 = 10_i128.pow(self.per_lp_base.abs().cast()?);
+
+        if self.per_lp_base > 0 {
+            AMM_RESERVE_PRECISION_I128.safe_mul(scalar)
+        } else {
+            AMM_RESERVE_PRECISION_I128.safe_div(scalar)
+        }
+    }
+
+    pub fn calculate_lp_delta(
```

Comment:
nit: could probably pass per_lp_base_unit into this and calculate_per_lp_delta to save some compute